app/src/main/java/com/example/lastfm/ArtistListFragment.java

```diff
@@ -0,0 +1,108 @@
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import io.reactivex.SingleObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class ArtistListFragment extends Fragment implements FragmentListener {
+    private final int NUMBER_OF_ARTISTS = 40;
+    private List<Artist> requestedArtists = new ArrayList<>();
+    private SwipeRefreshLayout mSwipeRefreshLayout;
+
+    private ArtistAdapter artistsAdapter = new ArtistAdapter(this::onArtistClick);
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        View view = inflater.inflate(R.layout.artist_list_fragment,null);
+        initRecyclerView();
+        retrofitRequest();
+        mSwipeRefreshLayout = view.findViewById(R.id.swiperefresh);
+        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                retrofitRequest();
+            }
+        });
+        return inflater.inflate(R.layout.artist_list_fragment, container, false);
+    }
+
+    private void retrofitRequest() {
+        LastFMClient client = ServiceGenerator.getInstance().getLastFMClient();
+        client.getArtists(NUMBER_OF_ARTISTS)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new SingleObserver<List<Artist>>() {
+                        @Override
+                        public void onSubscribe(Disposable d) {
+                        }
+
+                        @Override
+                        public void onSuccess(List<Artist> info) {
+                            requestedArtists = info;
+                            showArtists(requestedArtists);
+                            hideRefreshing();
+                        }
+
+                        @Override
+                        public void onError(Throwable e) {
+                            e.printStackTrace();
+                            hideRefreshing();
+                            String requestErrorText = getString(R.string.request_error_message);
+                            Toast.makeText(getActivity(), requestErrorText, Toast.LENGTH_LONG).show();
+                        }
+                    });
+    }
+
+    private void hideRefreshing(){
+        mSwipeRefreshLayout.setRefreshing(false);
+    }
+
+    private void showArtists(List<Artist> requestedArtists){
+        artistsAdapter.addItems(requestedArtists);
+    }
+
+    private void initRecyclerView() {
+        RecyclerView artistsRecyclerView = Objects.requireNonNull(getActivity()).findViewById(R.id.rv_artists);
```

Comment:
remove Objects.requireNonNull(getActivity())

Reply:
Changed to view.findviewbyid